pkg/lokifrontend/frontend/transport/handler.go

```diff
@@ -236,6 +236,7 @@ func writeError(w http.ResponseWriter, err error) {
 			err = errRequestEntityTooLarge
 		}
 	}
+	w.Header().Set("Content-Type", "text/plain; charset=UTF-8")
```

Comment:
Hhmm, I wonder if it's better for us to instead format the error as JSON so that at least we have some consistency.
What made you take this approach vs that one?

Reply:
Yeah, good point, I've also considered that, and thinking about it again, it's clearly the more consistent approach. I guess, people do not really depend on the response body being kept the same.
Also, I think there are still other occurrences where the response body is plain text. Will update the PR.

Reply:
Wouldn't that break the API? I'm not sure how strict we are there.

Reply:
A few thoughts:
- the API is inconsistent
- if we're changing anything it's technically breaking the API
- I'd rather break it in the direction of consistency :stuck_out_tongue_closed_eyes:

Reply:
Agree with consistency even by breaking the API response. But should definitely add it in our `docs/sources/upgrading/_index.md`. (/cc @chaudum )